src/pages/Models/Details/generators.js

```diff
@@ -56,12 +57,17 @@ export function generateIconImg(name, namespace) {
   );
 }
 
-export function generateEntityLink(namespace, href, name, subordinate) {
+export function generateEntityLink(namespace, name, subordinate) {
+  const charmStorePath = URL.fromAnyString(namespace)
+    .toString()
+    .replace("cs:", "");
   return (
     <>
       {subordinate && <span className="subordinate"></span>}
       {namespace && generateIconImg(name, namespace)}
-      <a href={href}>{name}</a>
+      <a data-test="app-link" href={`https://www.jaas.ai/${charmStorePath}`}>
```

Comment:
This will need to be a bit more resilient. If the charm is a local charm then we don't really want to try and generate a link. So your best bet might be to check and see if the `charm` value contains `cs:` first. If it doesn't then don't generate a link.